### ISSUE TITLE
differences between forks

### ISSUE THREAD

**Issue by flosse** [emk/duktape-rs#3]
There are three crates out there:
- https://crates.io/crates/duktape
- https://crates.io/crates/duktape_sys
- https://crates.io/crates/duktape_ffi

could you please explain the differences?

**Comment by muja** [comment]
@flosse, `duktape_sys` and `duktape_ffi` are the same - low-level bindings to the C functions from the original duktape -, the difference being that `duktape_ffi` compiles (on stable Rust). `duktape` is a crate that depends on `duktape_sys` and is a higher-level-wrapper with some abstractions and is still a WIP (that is discontinued as far as i can tell)

**Comment by emk** [comment]
I'm happy to transfer control of my crates (and the names on crates.io) to people actively working on duktape. Please don't hesitate to ask.

**Comment by moosingin3space** [comment]
I have a C project using Duktape right now, and I'd like nothing more than to port it to Rust. I'd be willing to take over `duktape-rs` if you are still looking for a maintainer.

**Comment by emk** [comment]
@flosse @moosingin3space Could you two perhaps work out a plan for who wants to be charge of maintaining this crate in the future? Thank you for any suggestions!

**Comment by moosingin3space** [comment]
@emk @flosse I went ahead and made a set of `bindgen`-generated bindings in my [duktape-sys](https://github.com/moosingin3space/duktape-sys) repository, which I'd like to become the `duktape-sys` crate on crates.io. At present, I'm also working on re-writing the Duktape REPL in Rust, and by attempting to minimize the amount of `unsafe` in that project, I am implementing a new high-level API for Duktape in Rust.

**Comment by moosingin3space** [comment]
Would everyone be okay with my `duktape-sys` repo taking over `duktape-sys` on crates.io?

**Comment by dflemstr** [comment]
@moosingin3space @emk I stumbled upon this repo after having created a `duktape-sys` of my own.  It looks like this: https://github.com/dflemstr/duk/tree/master/duktape-sys

I'm not sure which approach is best for this particular crate?

**Comment by moosingin3space** [comment]
@dflemstr I like your approach better.

**Comment by dflemstr** [comment]
@moosingin3space I think it would be nice to have a hybrid of sorts; since my code depends on libclang during build time it can't build on some platforms... Any ideas?

**Comment by dflemstr** [comment]
I opted to include the generated source code in the repository now and ship the code generator as a Cargo example... Might not be ideal but works rather well

**Comment by dflemstr** [comment]
@emk My duktape wrapper https://github.com/dflemstr/duk is now quite feature-ful, see the doc for details: https://dflemstr.github.io/duk/duk/

I would like to be able to upload it to crates.io but would hope to take over the `duktape` crate name for visibility.

**Comment by emk** [comment]
@dflemstr, your high-level wrappers look really useful.

What's the consensus here? I'm willing to consider turning over the name if everybody agrees on which crate should have it. :-)

**Comment by moosingin3space** [comment]
@emk I'm fine with @dflemstr taking the name.

**Comment by emk** [comment]
@dflemstr OK, I think the name is yours. Let me know how you want to handle the transition. I can give you commit rights on this repo, and access to the crate on crates.io.

**Comment by dflemstr** [comment]
Getting access on crates.io should be enough for now, thanks!

**Comment by emk** [comment]
Done! I've added you to `duktape`, `duktape_sys`, and this GitHub repository (the latter will at least allow you to see bugs, etc.) Thank you for working on Rust duktape, and my apologies for how long it has taken me to figure out what I was doing here.

**Comment by theduke** [comment]
@dflemstr seems like you never ended up working on this, right?

**Comment by dflemstr** [comment]
@theduke you are right, I kept using the duk name, I didn't find a reason to take over this specific crate
